src/FileFormats/LP/LP.jl

```diff
@@ -285,9 +285,429 @@ end
 #   Base.read!
 #
 # ==============================================================================
+const COMMENT_REG = r"(.*?)\\(.*)"
+function stripcomment(line::String)
+    if occursin("\\", line)
+        m = match(COMMENT_REG, line)
+        return strip(String(m[1]))
+    else
+        return strip(line)
+    end
+end
+
+# a list of section keywords in lower-case
+const KEYWORDS = Dict(
+    "max"      => Val{:obj},
+    "maximize" => Val{:obj},
+    "maximise" => Val{:obj},
+    "maximum"  => Val{:obj},
+    "min"      => Val{:obj},
+    "minimize" => Val{:obj},
+    "minimise" => Val{:obj},
+    "minimum"  => Val{:obj},
+
+    "subject to" => Val{:constraints},
+    "such that"  => Val{:constraints},
+    "st"         => Val{:constraints},
+    "s.t."       => Val{:constraints},
+
+    "bounds" => Val{:bounds},
+    "bound"  => Val{:bounds},
+
+    "gen"      => Val{:integer},
+    "general"  => Val{:integer},
+    "generals" => Val{:integer},
 
+    "bin"      => Val{:binary},
+    "binary"   => Val{:binary},
+    "binaries" => Val{:binary},
+
+    "end"      => Val{:quit}
+)
+
+const sense_alias = Dict(
+    "max"      => :Max,
+    "maximize" => :Max,
+    "maximise" => :Max,
+    "maximum"  => :Max,
+    "min"      => :Min,
+    "minimize" => :Min,
+    "minimise" => :Min,
+    "minimum"  => :Min
+)
+
+const subject_to_alias = ["subject to", "such that", "st", "s.t."]
+
+const CONSTRAINT_SENSE = Dict(
+    "<"  => :le,
+    "<=" => :le,
+    "="  => :eq,
+    "==" => :eq,
+    ">"  => :ge,
+    ">=" => :ge,
+)
+
+function verifyname(variable::String, maximum_length::Int)
+    if length(variable) > maximum_length
+        return false
+    end
+    # m = match(START_REG, variable)
+    # if !isnothing(m)
+    #     return false
+    # end
+    m = match(NAME_REG, variable)
+    if !isnothing(m)
+        return false
+    end
+    return true
+end
+
+# TODO Obj constant and SOS Variables
+mutable struct TempLPModel
```

Comment:
It's useful to ask whether we actually need this. Can't we just write into the `::Model` directly? I get that it's a much bigger rewrite, so perhaps we do this first to get the function and tests all sorted. Then we can fix it in another PR.

Reply:
I think that is probably the best way. Maybe we should also test if it is slow enough to encourage the rewrite.

Reply:
Yeah a good first step is what you have at the moment. Get all the tests and plumbing working, then focus on performance.